lisp/ess-inf.el

```diff
@@ -214,23 +233,11 @@ This may be useful for debugging."
                  (switches
                   (when (and switches-symbol (boundp switches-symbol))
                     (symbol-value switches-symbol))))
-
-            (inferior-ess-mode)
+            (ess--set-inferior-major-mode ess-dialect)
             (ess-write-to-dribble-buffer
              (format "(inf-ess 3.0): prog=%s, start-args=%s, echoes=%s\n"
                      inferior-ess-program infargs comint-process-echoes))
             (setq ess-local-process-name procname)
-            (goto-char (point-max))
-
-            (when ess-history-file
-              ;; Load past history
```

Comment:
Why this history-file treatment was removed?

Reply:
Nevermind. You moved it to ess-inferior-mode. I think it's a good idea.

Reply:
It's not gone - it's in `inferior-ess-mode` now. I tried to make when variables are set a little more standard (it's still not perfect) by setting as many as possible in the major mode setup. It makes more sense (to me) to have it in inferior-ess-mode rather than inferior-ess since it has nothing to do with the actual process (i.e. we never query the process or send anything to it).